### PR TITLE
Makes More Objects Climbable

### PR DESCRIPTION
**Pull request by SleepyGemmy**
this PR makes gas canisters and closets climbable.

## changes
- makes gas canisters climbable.
- makes closets and lockers climbable, excluding wall mounted ones.
- adds a `climb_time` variable to objects so each object can have its own required time to climb it.
- makes all `/obj` climbable, if the variable is set to `TRUE`.
- adds a dedicated `shakable` variable to objects, for things that can't be shaken, like anchored or heavy structures/machinery.

### CODE REVIEW

**Comment by SleepyGemmy** [comment]
!review

**Review by alsoandanswer — COMMENTED** [review]
Could just get rid of the climbable variable. You can combine it with climb_time, by making it so anything with a climb_time of 0 or -1 be unclimbable.

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -258,3 +264,159 @@
...
+					affecting.parent.add_autopsy_data("Misadventure", damage)
+			else
+				to_chat(H, SPAN_DANGER("You land heavily!"))
+				H.adjustBruteLoss(damage)
+
+			H.UpdateDamageIcon()
+			H.updatehealth()
+	return
+
+/obj/proc/can_touch(var/mob/user)
```
Comment (alsoandanswer) [inline code/game/objects/objs.dm:409]: use_check_and_message() can probably be used here instead

File: code/game/objects/objs.dm [hunk 2]
```diff
@@ -258,3 +264,159 @@
...
+//
+
+/obj/attack_hand(mob/user)
+	if(shakable)
+		if(LAZYLEN(climbers) && !(user in climbers))
+			user.visible_message(
+				SPAN_DANGER("\The [user] shakes \the [src]."),
+				SPAN_NOTICE("You shake \the [src].")
+			)
+		shake_object()
```
Comment (NonQueueingMatt) [inline code/game/objects/objs.dm:279]: this is missing a ..()

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Comment by SleepyGemmy** [comment]
closing for now.

**Comment by SleepyGemmy** [comment]
not pursuing this.

**Pull request closed without merging**